stingray/covariancespectrum.py

```diff
@@ -0,0 +1,194 @@
+                toa = event_list_T[0][np.logical_and(
+                    event_list_T[1] >= energy - least_count*0.5,
+                    event_list_T[1] < energy + least_count*0.5)]
+                self.energy_events[energy] = sorted(toa)
+
+    def _update_energy_events(self):
+        """
+        In case of a specific band interest, merge the required energy bins
+        into one with the new key as the mid-point of the band interest.
+        """
+        if self.band_interest is not None:
+            energy_events_ = {}
+            for band in list(self.band_interest):
+                mid_bin = (band[0] + band[1]) / 2
+                energy_events_[mid_bin] = []
+
+                # Modify self.energy_events to form a band with one key
+                for key in list(self.energy_events.keys()):
+                    if key >= band[0] and key <= band[1]:
+                        energy_events_[mid_bin] += self.energy_events[key]
+                        del self.energy_events[key]
+
+            self.energy_events.update(energy_events_)
+
+    def _init_energy_covar(self):
+        """
+        Initialize the energy_covar dictionary for further computations.
+        """
+        # The dictionary with covariance spectrum for each energy bin
+        self.energy_covar = {}
+
+        # Initialize it with empty mapping
+        if self.band_interest is None:
+            for key in self.energy_events.keys():
+                self.energy_covar[key] = []
+        else:
+            for band in list(self.band_interest):
+                mid_bin = (band[0] + band[1]) / 2
+                self.energy_covar[mid_bin] = []
+
+    def _construct_energy_covar(self):
+        """Form the actual output covaraince dictionary and array."""
+        self._init_energy_covar()
+
+        for energy in self.energy_covar.keys():
+            lc = Lightcurve.make_lightcurve(
+                    self.energy_events[energy], self.dt, tstart=self.min_time,
+                    tseg=self.max_time - self.min_time)
+
+            # Calculating timestamps for lc_ref
+            toa_ref = []
+            for key, value in self.energy_events.items():
+                if key >= self.ref_band_interest[0] and \
+                        key <= self.ref_band_interest[1]:
+                    if key != energy:
+                        toa_ref.extend(value)
+
+            toa_ref = np.array(sorted(toa_ref))
+
+            lc_ref = Lightcurve.make_lightcurve(
+                    toa_ref, self.dt, tstart=self.min_time,
+                    tseg=self.max_time - self.min_time)
+
+            assert len(lc.time) == len(lc_ref.time)
+
+            covar = self._compute_covariance(lc, lc_ref)
+
+            self.energy_covar[energy] = covar
+
+        self.covar = np.vstack(self.energy_covar.items())
+
+    def _compute_covariance(self, lc1, lc2):
+        """Calculate and return the covariance between two time series."""
+        return np.cov(lc1.counts, lc2.counts)[0][1]
```

Comment:
Hmmm, why did you pick element `[0][1]` here? Based on [Wilkinson & Uttley 2009](http://mnras.oxfordjournals.org/content/397/2/666.full.pdf), Equation 2, it seems to me that this might be the sum of the diagonal elements of `numpy.cov`, but I might be confused.